src/builders/polygons.js

```diff
@@ -7,64 +7,80 @@ import earcut from 'earcut';
 
 const up_vec3 = [0, 0, 1];
 
-// Tesselate a flat 2D polygon
-// x & y coordinates will be set as first two elements of provided vertex_template
+
+/**
+ * To tesselate flat 2D polygons.
+ * The x & y coordinates will be set as first two elements of provided vertex_template.
+ * @param {Array.<Array.<Array.<Array.<number>>>>} polygons The polygons to tesselate.
+ * @param {VertexData} vertex_data The VertexData were to store the results
+ * @param {Array.<number>} vertex_template The vertex template to use
+ * @param {Object} options The texture coordinate options to apply
+ * @return {number} the number of the resulting geometries (triangles)
+ */
 export function buildPolygons (
-    polygons,
-    vertex_data, vertex_template,
-    { texcoord_index, texcoord_scale, texcoord_normalize }) {
+        polygons, vertex_data, vertex_template, { texcoord_index, texcoord_scale, texcoord_normalize }) {
 
-    var vertex_elements = vertex_data.vertex_elements;
+    let vertex_elements = vertex_data.vertex_elements,
+        element_offset = vertex_data.vertex_count,
+        num_polygons = polygons.length,
+        geom_count = 0;
 
     if (texcoord_index) {
         texcoord_normalize = texcoord_normalize || 1;
+        // Usage of "var" for function wide visibility:
```

Comment:
If I recall, the use of `var` was to get around a limitation in how babel was parsing destructuring, where it was not allowing me to define them once and then reassign with just `[min_u, min_v, max_u, max_v] = texcoord_scale || default_uvs;`. But I think this may be fixed, in which case we could use `let`.

Reply:
We can't use let here, if the variable is used outside the if block. Of course, as long as babel is used it will work, but not without babel.

Reply:
That makes sense. I was thinking define it above with the others (e.g. [here](https://github.com/tangrams/tangram/pull/651/files/c4652f00818fde764867f23ad369a5ef45eb6eed#diff-052edb1da57e539c22cf5308cd97bf7eR23)). They are only then used if the geometry has UV coords.

Reply:
Ahh, I just mistaken it, will do …